dsc_lib/src/configure/config_doc.rs

```diff
@@ -30,7 +30,11 @@ pub enum Operation {
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, JsonSchema)]
 pub enum ExecutionKind {
     Actual,
-    WhatIf,
+    // differentiate internally whether what-if should be processed by the resource or dsc engine
+    #[serde(rename = "WhatIf")]
+    WhatIfDSC,
+    #[serde(rename = "WhatIf")]
+    WhatIfResource,
```

Comment:
Do we need to differentiate between config and resource what-if?

Reply:
Not sure if this answers the question but this is differentiating whether the `what-if` should be processed synthetically by DSC or whether the resource itself implements `what-if`

Reply:
I see.  I don't think we need to explicitly differentiate just like we don't indicate if `test` is implemented or synthetic.  Is there a specific scenario where you think this is useful?